### PR TITLE
Create page on API documentation with Swagger

### PR DESCRIPTION
**Pull request by dianzrong**
I edited 2 files: one of the article I wrote, and the development process file which I changed to add the link to the article mentioned above. 

Feedback, especially on the tone, formality, and level of prior knowledge assumed (i.e. discussing endpoints without defining them) would be appreciated.

### CODE REVIEW

File: Topics/Development_Process/Documentation/Swagger_API_Documentation.md [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+## Table of Contents
+- [Purpose of API Documentation](#purpose-of-api-documentation)
+- [Keys to Good API Documentation](#keys-to-good-api-documentation)
+- [Introduction of SwaggerHub](#introduction-of-swaggerhub)
+  - [Features](#features)
+- [Other Free Tool](#other-free-tool)
+
+
+## Purpose of API Documentation
+As the name suggests, API documentation describes a handbook of instructions and expectations around working with the project's APIs.
```
Comment (foyarekhua) [inline Topics/Development_Process/Documentation/Swagger_API_Documentation.md:12]: maybe give a quick description of what an API is. I'm not too sure a new csc301 student would know what that is

File: Topics/Development_Process/Documentation/Swagger_API_Documentation.md [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+- cover all the basics of each endpoint:
+  - the path of each endpoint
+  - what methods each endpoint supports
+  - what is accepted and/or what is returned from each endpoint
+    - including sample responses and errors
+- update the documentation when changes are made to the code
+- consistent format to allow easy access to relevant information
+
+## Introduction of SwaggerHub
+[SwaggerHub](https://swagger.io/tools/swaggerhub/) is one tool created to make API documentation easy to edit and read.
```
Comment (foyarekhua) [inline Topics/Development_Process/Documentation/Swagger_API_Documentation.md:28]: if you want, you can probably expand more on this. Like why use swaggerhub over something else. What makes this special? Y not google doc

Comment (foyarekhua) [reply Topics/Development_Process/Documentation/Swagger_API_Documentation.md:28]: wait never mind I see you expand it in the features section

File: Topics/Development_Process/Documentation/Swagger_API_Documentation.md [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+
+## Introduction of SwaggerHub
+[SwaggerHub](https://swagger.io/tools/swaggerhub/) is one tool created to make API documentation easy to edit and read. 
+
+### Features
+- easy to edit: can use code or visual editor on their website and undo changes to previous revisions
+- easy to learn: there are many resources, [example](https://support.smartbear.com/swaggerhub/docs/en/get-started/basics-of-swaggerhub.html)
+- easy to share: by default, the API documentation is accessible by a public link, or the contents can be exported to a html file, json, etc.
+- easy to organize: can group endpoints and methods by tags
+- easy to standardize: can create schema for what typical responses may be
```
Comment (foyarekhua) [inline Topics/Development_Process/Documentation/Swagger_API_Documentation.md:35]: maybe expand on this point. specifically what we are standardizing. Or give an example

File: Topics/Development_Process/Documentation/Swagger_API_Documentation.md [hunk 4]
```diff
@@ -0,0 +1,44 @@
...
+- easy to iterate: can create multiple versions, fork, and even merge API documents
+- interactive: can generate sample requests and outputs from the link to the API
+- free: the above features can all be accessed for free, however adding a collaborator requires the enteprise plan
+
+## Other Free Tools
```
Comment (foyarekhua) [inline Topics/Development_Process/Documentation/Swagger_API_Documentation.md:40]: I like that this section was added :)

**Comment by foyarekhua** [comment]
Overall great job on introducing swaggerhub and its purpose. I will not become swaggerhub pro 😎

File: Topics/Development_Process/Documentation/Swagger_API_Documentation.md (hunk 1 above)
Comment (dianzrong) [reply Topics/Development_Process/Documentation/Swagger_API_Documentation.md:23]: Addressed [here](https://github.com/learning-software-engineering/learning-software-engineering.github.io/pull/279/commits/374c16346a04a489f48b529dd5db62113158b6bc)

Comment (dianzrong) [reply Topics/Development_Process/Documentation/Swagger_API_Documentation.md:23]: I tried to keep it brief so I only explained the parts that were relevant

File: Topics/Development_Process/Documentation/Swagger_API_Documentation.md (hunk 3 above)
Comment (dianzrong) [reply Topics/Development_Process/Documentation/Swagger_API_Documentation.md:45]: That's a good point. I forgot that there are many things in APIs that should be standardized

Comment (dianzrong) [reply Topics/Development_Process/Documentation/Swagger_API_Documentation.md]: Addressed in [9ec6a35](https://github.com/learning-software-engineering/learning-software-engineering.github.io/pull/279/commits/9ec6a354e2ccb3c4a51d5b73e4beede107dd0f00)

File: Topics/Development_Process/Documentation/Swagger_API_Documentation.md (hunk 1 above)
Comment (foyarekhua) [reply Topics/Development_Process/Documentation/Swagger_API_Documentation.md:23]: Small typo. Says "andd" but other than that looks good!

Comment (foyarekhua) [reply Topics/Development_Process/Documentation/Swagger_API_Documentation.md:23]: this was meant for the comment about standardizing. The typo is on line 46

Comment (foyarekhua) [reply Topics/Development_Process/Documentation/Swagger_API_Documentation.md:23]: > Addressed [here](https://github.com/learning-software-engineering/learning-software-engineering.github.io/pull/279/commits/374c16346a04a489f48b529dd5db62113158b6bc)

I think this was a great solution! It goes into just enough detail to provide context but not too much where it's overwhelming or borderline off topic

**Comment by foyarekhua** [comment]
The organisation was well thought out from the table of content to how you formatted your subtopics. The content itself has the right amount of detail making it an easy read (an important thing to consider when introducing a new topic to a CSC301 student like me!). After fixing the typo, you can merge!

**Review by foyarekhua — APPROVED** [review]

**Pull request merged**